packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts

```diff
@@ -14,30 +14,34 @@
  * limitations under the License.
  */
 
-import {AggregationType, DistributionData, logger, Logger, Measure, Measurement, MeasureType, MeasureUnit, StatsEventListener, Tags, View} from '@opencensus/core';
+import {logger, Logger, Measurement, Metric, MetricDescriptor as OCMetricDescriptor, MetricProducerManager, Metrics, StatsEventListener, View} from '@opencensus/core';
 import {auth, JWT} from 'google-auth-library';
 import {google} from 'googleapis';
-import * as path from 'path';
 
-import {Distribution, LabelDescriptor, MetricDescriptor, MetricKind, Point, StackdriverExporterOptions, TimeSeries, ValueType} from './types';
+import {createMetricDescriptorData, createTimeSeriesList} from './stackdriver-stats-utils';
+import {StackdriverExporterOptions, TimeSeries} from './types';
 
 google.options({headers: {'x-opencensus-outgoing-request': 0x1}});
 const monitoring = google.monitoring('v3');
-
-const RECORD_SEPARATOR = String.fromCharCode(30);
-const UNIT_SEPARATOR = String.fromCharCode(31);
+const GOOGLEAPIS_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
 
 /** Format and sends Stats to Stackdriver */
 export class StackdriverStatsExporter implements StatsEventListener {
   private period: number;
   private projectId: string;
   private metricPrefix: string;
-  private viewToUpload:
-      {[key: string]: {view: View; tags: {[key: string]: Tags;}}};
+  private displayNamePrefix: string;
+  // tslint:disable:no-any
+  private onMetricUploadError: any;
```

Comment:
It seems like a possible value of this is a function. Could you make the type something like `(err: Error) => void|{}`, which would allow it to either a function that takes an error, or just any arbitrary object? That way there is some amount of documentation about what role it plays.

Reply:
Agreed, done.